node/src/types/block/hot_block/state.rs

```diff
@@ -0,0 +1,250 @@
+use datasize::DataSize;
+use serde::Serialize;
+
+use super::MergeMismatchError;
+
+#[derive(Clone, Copy, Debug, DataSize)]
+pub(crate) enum StateChange {
+    Updated,
+    AlreadyRegistered,
+}
+
+impl StateChange {
+    pub(crate) fn was_updated(self) -> bool {
+        matches!(self, StateChange::Updated)
+    }
+
+    pub(crate) fn was_already_registered(self) -> bool {
+        matches!(self, StateChange::AlreadyRegistered)
+    }
+}
+
+impl From<bool> for StateChange {
+    fn from(current_state: bool) -> Self {
+        if current_state {
+            StateChange::AlreadyRegistered
+        } else {
+            StateChange::Updated
+        }
+    }
+}
+
+#[derive(Clone, Copy, Eq, PartialEq, Default, Serialize, Debug, DataSize)]
+pub(crate) struct State {
+    pub(super) immediate_switch_block_for_current_protocol_version: bool,
+    pub(super) stored: bool,
+    pub(super) sent_to_deploy_buffer: bool,
+    pub(super) updated_validator_matrix: bool,
+    pub(super) gossiped: bool,
+    pub(super) executed: bool,
+    pub(super) tried_to_sign: bool,
+    pub(super) sent_to_consensus_post_execution: bool,
+    pub(super) sent_to_accumulator_post_execution: bool,
+    pub(super) sufficient_finality: bool,
+    pub(super) marked_complete: bool,
+}
+
+impl State {
+    /// Returns a new `State` with all fields set to `false`.
+    pub(crate) fn new() -> Self {
+        Self::default()
+    }
+
+    /// Returns a new `State` with all fields set to `false` except for
+    /// `is_immediate_switch_block_for_current_protocol_version`.
```

Comment:
Nit:

```suggestion
    /// `immediate_switch_block_for_current_protocol_version`.
```

Reply:
Fixed in [e4d8cc6](https://github.com/casper-network/casper-node/pull/3549/commits/e4d8cc609a03671e5923825bf06174f6c64e9e7f).